crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -0,0 +1,255 @@
+pub use bevy_hecs::SystemParam;
+
+use crate::{
+    ChangedRes, Commands, FromResources, Local, Query, QuerySet, QueryTuple, Res, ResMut, Resource,
+    ResourceIndex, Resources, SystemState,
+};
+use bevy_hecs::{ArchetypeComponent, Fetch, Or, Query as HecsQuery, TypeAccess, World};
+use parking_lot::Mutex;
+use std::{any::TypeId, sync::Arc};
+
+pub trait SystemParam: Sized {
+    fn init(system_state: &mut SystemState, world: &World, resources: &mut Resources);
+    /// # Safety
+    /// This call might access any of the input parameters in a safe way. Make sure the data access is safe in
+    /// the context of the system scheduler
+    unsafe fn get_param(
```

Comment:
I'm not sure what guarantee this `unsafe fn` makes. If the implementation "might access any of the input parameters in a safe way", then isn't it safe to call?

Reply:
Bawhaha oops its definitely supposed to say "unsafe" there. Nice catch